lib/http/HttpRequestEncoder.cpp

```diff
@@ -45,6 +45,11 @@ namespace MAT_NS_BEGIN {
         ctx->httpRequest->SetUrl(m_config.GetCollectorUrl());
 
         ctx->httpRequest->GetHeaders().set("Expect", "100-continue");
+
+#ifdef HAVE_MAT_EVT_TRACEID 
+        ctx->httpRequest->GetHeaders().set("Trace-Id", ctx->traceId);
```

Comment:
Can we add a simple test for this with macro define to ensure we get this field in header(future proofing change)

Reply:
Ack. will address this comment in PR for main 1DS repo